src/workerd/api/util.c++

```diff
@@ -284,4 +285,70 @@ double dateNow() {
   return 0.0;
 }
 
+kj::Maybe<jsg::V8Ref<v8::Object>> cloneRequestCf(
+    jsg::Lock& js,
+    kj::Maybe<jsg::V8Ref<v8::Object>> maybeCf) {
+  KJ_IF_MAYBE(cf, maybeCf) {
+    // In case the cf object has a logging proxy, we want to make sure
+    // the logging is not triggered here when the object is cloned.
+    NoRequestCfProxyLoggingScope noLoggingScope;
+    auto cloned = cf->deepClone(js);
+    maybeWrapBotManagement(js.v8Isolate, cloned.getHandle(js));
+    return kj::mv(cloned);
+  }
+  return nullptr;
+}
+
+void maybeWrapBotManagement(v8::Isolate* isolate, v8::Local<v8::Object> handle) {
+  auto context = isolate->GetCurrentContext();
+  auto botManagement = jsg::v8StrIntern(isolate, "botManagement");
+  v8::Local<v8::Value> maybeBotManagement = jsg::check(handle->Get(context, botManagement));
+  // If the botManagement field exists, we replace its value here with a Proxy object that
+  // will log the first time any of its properties are accessed. Logging will occur only
+  // once per worker instance.
+  //
+  // Replacing the value with a proxy rather than setting an accessor for botManagement on
+  // the request.cf object itself avoids false positives when someone is simply iterating
+  // over the fields of request.cf without actually using them. It also allows us to avoid
+  // having to create accessors or a class to intercept every individual property on the
+  // botManagement object.
+  if (maybeBotManagement->IsObject() && !maybeBotManagement->IsProxy()) {
```

Comment:
Hang on, can we make this behavior conditional on `botsManagement.subscription` being false, and ask the team to start sending that field? That way we don't get any noisy logs for people who are legitimately subscribed.

Reply:
Ok, added.